### PR TITLE
Support remove fabric for access control extension

### PR DESCRIPTION
**Pull request by mlepage-google**
#### Problem
Extension attribute needs its fabric scoped data removed
when the fabric is removed.

Part of issue #10252

#### Change overview
Add fabric table delegate to listen for fabric removal notifications.

Use them to remove the fabric-scoped extension data.

#### Testing
- Ran chip-all-clusters-app with chip-tool on Linux
- Added extension, verified data present in /tmp/chip_kvs
- Removed fabric, verified data absent in /tmp/chip_kvs

### CODE REVIEW

**Comment by mlepage-google** [comment]
Needs to be rebased/built off PR #17008

**Review by woody-apple — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Review by cecille — APPROVED** [review]

File: src/app/clusters/access-control-server/access-control-server.cpp [hunk 1]
```diff
@@ -705,9 +705,27 @@ CHIP_ERROR AccessControlAttribute::WriteExtension(const ConcreteDataAttributePat
...
+
 } // namespace
 
 void MatterAccessControlPluginServerInitCallback()
 {
+    ChipLogProgress(DataManagement, "AccessControlCluster: initializing");
+
     registerAttributeAccessOverride(&gAttribute);
+
+    Server::GetInstance().GetFabricTable().AddFabricDelegate(&fabricTableDelegate);
```
Comment (tcarmelveilleux) [inline src/app/clusters/access-control-server/access-control-server.cpp:730]: How many slots are there of these? Could we run out and not notice due to not being able to return errors from this init?

**Review by tcarmelveilleux — APPROVED** [review]

File: src/app/clusters/access-control-server/access-control-server.cpp (hunk 1 above)
Comment (mlepage-google) [reply src/app/clusters/access-control-server/access-control-server.cpp:730]: Linked list.

**Review by msandstedt — APPROVED** [review]

**Pull request merged**